src/Rocketeer/Tasks/Update.php

```diff
@@ -16,39 +16,49 @@
  */
 class Update extends Deploy
 {
-	 /**
-	 * A description of what the Task does
+	/**
+	 * A description of what the task does
 	 *
 	 * @var string
 	 */
 	protected $description = 'Update the remote server without doing a new release';
 
 	/**
-	 * Run the Task
+	 * Run the task
 	 *
-	 * @return  void
+	 * @return  boolean|null
 	 */
 	public function execute()
 	{
+		// Check if local is ready for deployment
+		if (!$this->executeTask('Primer')) {
+			return $this->halt('Project is not ready for deploy. You were almost fired.');
+		}
+
 		// Update repository
-		$this->updateRepository();
+		if (!$this->getStrategy('Deploy')->update()) {
+			return $this->halt();
+		}
 
 		// Recreate symlinks if necessary
-		$this->syncSharedFolders();
+		$this->steps()->syncSharedFolders();
```

Comment:
2.1 will bring tasks and strategies options and among them is a `shared_dependencies` option for the Dependencies task that automatically shares the dependencies from one release to another, having a lot of the time. You can also set it to `copy` to preserve isolation, which will copy the dependencies folders from the previous release, and then update them